src/main/java/com/iota/iri/service/milestone/impl/MilestoneServiceImpl.java

```diff
@@ -160,10 +188,10 @@ public void resetCorruptedMilestone(int index) throws MilestoneException {
     }
 
     @Override
-    public MilestoneValidity validateMilestone(TransactionViewModel transactionViewModel, int milestoneIndex,
-            SpongeFactory.Mode mode, int securityLevel) throws MilestoneException {
+    public MilestoneValidity validateMilestone(TransactionViewModel transactionViewModel, int milestoneIndex)
+            throws MilestoneException {
 
-        if (milestoneIndex < 0 || milestoneIndex >= 0x200000) {
+        if (milestoneIndex < 0 || milestoneIndex >= maxMilestoneIndex) {
```

Comment:
this will allow higher depths of trees. (and tighter validation for lower trees)